provider/diag.go

```diff
@@ -0,0 +1,22 @@
+// Copyright 2016-2023, Pulumi Corporation.
+
+package provider
+
+import (
+	"github.com/hashicorp/aws-sdk-go-base/v2/diag"
+)
+
+// formatDiags converts AWS SDK Diagnostics into a human-readable string.
+func formatDiags(diag diag.Diagnostics) string {
+	details := ""
```

Comment:
[nit] `strings.Builder`? Not worth it here?

Reply:
I'd say it would not be noticeable for end-users unless something pushes MBs into it.

Reply:
Changed to strings.Builder. I think it's reasonable readable and overall good hygiene, thank you for the suggestion.